### PR TITLE
fixes #100

### LINKED ISSUE #100: GET STIG ccis endpoint returning 500 errors
**Bug Location**

I found a bug in the:
- [x] API
- [ ] UI
- [ ] Elsewhere (Please specify)

**Bug Description:**
500 Internal Server Error response to:
GET stigs/ccis/{cci}
GET stigs/{benchmarkId}/revisions/{revisionString}/ccis

TEST NOTES:
Tests exist for this endpoint, but currently commented out in block starting with:
"START AZDO #117"

### PR DESCRIPTION
**Pull request by russdj**
Fixes #100 
implements getCci and getCcisByRevision

### CODE REVIEW

File: api/source/service/mysql/STIGService.js [hunk 1]
```diff
@@ -949,10 +949,90 @@ exports.deleteStigById = async function(benchmarkId, userObject) {
...
+  if ( inProjection && inProjection.includes('emassAp') ) {
+    columns.push(`case when c.apAcronym is null then null else json_object("apAcronym", c.apAcronym, "implementation", c.implementation, "assessmentProcedure", c.assessmentProcedure) END  as "emassAp"`)
+  }
+
+  if ( inProjection && inProjection.includes('references') ) {
+    columns.push(`(select 
+      coalesce
+      (
+        (
+          select json_arrayagg (json_object(
```
Comment (csmig) [inline api/source/service/mysql/STIGService.js:984]: Please include the other properties from the OAS definition:
```
        "version": "string",
        "location": "string",
        "indexDisa": "string",
        "textRefNist": "string",
        "parentControl": "string"
```
these are all in `crm` which is handy!

File: api/source/service/mysql/STIGService.js [hunk 2]
```diff
@@ -969,7 +1049,34 @@ exports.getCci = async function(cci, userObject) {
...
-    let rows = await _this.METHOD()
+
+    let [input, version, release] = /V(\d+)R(\d+(\.\d+)?)/.exec(revisionStr)
+
+    let sql = `
+      SELECT DISTINCT
+          c.cci,
+          c.type,
+        COALESCE((
+          SELECT JSON_ARRAYAGG(JSON_OBJECT(
```
Comment (csmig) [inline api/source/service/mysql/STIGService.js]: Same as above. Please include the other properties from the OAS definition:
```
        "version": "string",
        "location": "string",
        "indexDisa": "string",
        "textRefNist": "string",
        "parentControl": "string"
```
these are all in `crm` which is handy!

File: api/source/service/mysql/STIGService.js (hunk 1 above)
Comment (russdj) [reply api/source/service/mysql/STIGService.js:984]: added

File: api/source/service/mysql/STIGService.js (hunk 2 above)
Comment (russdj) [reply api/source/service/mysql/STIGService.js]: added

File: api/source/service/mysql/STIGService.js [hunk 3]
```diff
@@ -969,7 +1054,39 @@ exports.getCci = async function(cci, userObject) {
  **/
 exports.getCcisByRevision = async function(benchmarkId, revisionStr, userObject) {
   try {
-    let rows = await _this.METHOD()
+
+    let [input, version, release] = /V(\d+)R(\d+(\.\d+)?)/.exec(revisionStr)
```
Comment (csmig) [inline api/source/service/mysql/STIGService.js]: Unfortunately this approach, which is incorrectly present elsewhere in this file, is insufficient. It throws if revisionStr == "latest", which is a legal value per the the OAS.

The various service methods handle "latest" in various ways. In this case, please consider using the approach in STIGService.js line 79. 
```
  if (inPredicates.revisionStr != 'latest') {
    joins = ['revision r']
    let [results, version, release] = /V(\d+)R(\d+(\.\d+)?)/.exec(inPredicates.revisionStr)
    predicates.statements.push('r.version = ?')
    predicates.binds.push(version)
    predicates.statements.push('r.release = ?')
    predicates.binds.push(release)
  } else {
    joins = ['current_rev r']
  }
```
This approach does require this query be decomposed into arrays for columns, joins, predicates. This must be the only complex query in the API that is not constructed using that pattern

Comment (csmig) [reply api/source/service/mysql/STIGService.js]: @cd-rite Giving you a heads up that the newman test does not make a request to this endpoint using "latest", so this code is passing when it should not.

Comment (russdj) [reply api/source/service/mysql/STIGService.js]: reworked to add handling of latest

**Comment by csmig** [comment]
I reverted the return statement to return empty arrays, to remain consistent with how the API is currently handling this. The idea of returning 204 when no rows exists is worth considering, but not here and now.

**Pull request merged**